specs/kiota.config.md

```diff
@@ -0,0 +1,634 @@
+              },
+            },*/
+            "serializers": [
+              "Contoso.Json.CustomSerializer"
+            ],
+            "deserializers": [
+              "Contoso.Json.CustomDeserializer"
+            ],
+            "structuredMimeTypes": [
+              "application/json"
+            ],
+            "usesBackingStore": true,
+            "includeAdditionalData": true
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+#### Using kiota generate with parameters inferred from the kiota.config file
+
+```bash
+kiota generate
+```
+
+#### Using kiota generate with parameters inferred from the kiota.config file for a single API
+
+```bash
+kiota generate --api-name "graph" --client-name "graphDelegated"
+```
+
+#### Using kiota generate with parameters inferred when there are no kiota.config file
+
+```bash
+kiota generate --openapi "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml" --include-path "/me/chats#GET" --include-path "/me#GET" --language csharp --class-name "GraphClient" --namespace-name "Contoso.GraphApp" --backing-store --exclude-backward-compatible --serializer "Contoso.Json.CustomSerializer" --deserializer "Contoso.Json.CustomDeserializer" -structured-mime-types "application/json" --output "./generated/graph/csharp"
+```
+
+```javascript
+// This file gets generated and then `kiota generate` is executed based on these parameters
+{
+  "name": "Contoso.GraphApp", // Inferred from the provided --namespace-name or its default value
+  "apis": {
+    "https://graph.microsoft.com/v1.0": { // Inferred from the first server entry in the OpenAPI description
+      "descriptionHash": "9EDF8506CB74FE44...",
+      "descriptionLocation": "https://.../openapi.yaml",
+      "includePatterns": ["/me/chats#GET", "/me#GET"],
+      "excludePatterns": [],
+      "clients": {
+        "GraphClient": { // Inferred from the provided --class-name or its default value 
+          "language": "csharp",
+          "outputPath": "./generated/graph/csharp",
+          "clientClassName": "GraphClient",
+          "clientNamespaceName": "Contoso.GraphApp",
+          "features": {
+            // Adding for future visibility, but not required for now
+            /*"authentication": {
+              "authenticationProvider": "Microsoft.Kiota.Authentication.AzureAuthProvider",
+              "authenticationParameters": {
+                "clientId": "guid"
+              },
+            },*/
+            "serializers": [
+              "Contoso.Json.CustomSerializer"
+            ],
+            "deserializers": [
+              "Contoso.Json.CustomDeserializer"
+            ],
+            "structuredMimeTypes": [
+              "application/json"
+            ],
+            "usesBackingStore": true,
+            "includeAdditionalData": true
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+## End-to-end scenarios using the CLI
+
+### Get started to generate an API
+
+```bash
```

Comment:
in this proposal, I'm not seeing any mention of the update command. 
We could reuse it and have and additional parameter like `--maintain-descriptions` (default false)
This way:

# generate for a single API

Does:
1. init (no new command for that, add implicitly does it for you)
2. add api/client
3. update

# update (without the new parameter)

refreshes the client with the new version of kiota, and refreshes the description while at it.

# update (with the new parameter)

only refreshes the client, not the descriptions